pkg/geoipupdate/internal/retry.go

```diff
@@ -20,36 +19,29 @@ func MaybeRetryRequest(c *http.Client, retryFor time.Duration, req *http.Request
 	if req.Body != nil {
 		return nil, errors.New("can't retry requests with bodies")
 	}
-	exp := backoff.NewExponentialBackOff()
-	exp.MaxElapsedTime = retryFor
 	var resp *http.Response
-	err := backoff.Retry(
-		func() error {
-			if resp != nil {
-				r := resp
-				resp = nil
-				if err := r.Body.Close(); err != nil {
-					return errors.Wrap(err, "error closing response body before retrying")
-				}
-			}
-			var err error
-			resp, err = c.Do(req) // nolint: bodyclose
-			if err != nil {
-				return errors.Wrap(err, "error performing http request")
-			}
-			if resp.StatusCode/100 == 5 {
-				return &internalServerError{}
+	var err error
+
+	start := time.Now()
+	for i := uint(0); ; i++ {
+		resp, err = c.Do(req)
+		if err == nil {
+			if resp.StatusCode < 500 {
+				break
 			}
-			return nil
-		},
-		exp,
-	)
-	if _, ok := err.(*internalServerError); ok {
-		return resp, nil
-	}
-	return resp, err
-}
+			_ = resp.Body.Close()
```

Comment:
If we break due to exceeding our time and we have an HTTP 5xx, will the reads of the body we try to do in the caller error in an unexpected way?

Reply:
Good thought. I changed the behavior on it. I force pushed the changes.